src/Microsoft.PowerShell.Commands.Management/commands/management/NewPropertyCommand.cs

```diff
@@ -175,4 +182,124 @@ protected override void ProcessRecord()
         #endregion Command code
 
     }
+
+    /// <summary>
+    /// Provides argument completion for PropertyType parameter.
+    /// </summary>
+    public class PropertyTypeArgumentCompleter : IArgumentCompleter
+    {
+        private static readonly string[] s_RegistryPropertyTypes = new string[]
+        {
+            "String",
+            "ExpandString",
+            "Binary",
+            "DWord",
+            "MultiString",
+            "QWord",
+            "Unknown"
+        };
+
+        private static string GetRegistryPropertyTypeToolTip(string propertyTypeName) => propertyTypeName switch
+        {
+            "String" => TabCompletionStrings.RegistryStringToolTip,
+            "ExpandString" => TabCompletionStrings.RegistryExpandStringToolTip,
+            "Binary" => TabCompletionStrings.RegistryBinaryToolTip,
+            "DWord" => TabCompletionStrings.RegistryDWordToolTip,
+            "MultiString" => TabCompletionStrings.RegistryMultiStringToolTip,
+            "QWord" => TabCompletionStrings.RegistryQWordToolTip,
+            _ => TabCompletionStrings.RegistryUnknownToolTip
+        };
+
+        private static readonly char[] QuoteChars = new char[]
+        {
+            '\'',
+            '"',
+            '\u2018', // left single quotation mark
+            '\u2019', // right single quotation mark
+            '\u201a', // single low-9 quotation mark
+            '\u201b', // single high-reversed-9 quotation mark
+            '\u201c', // left double quotation mark
+            '\u201d', // right double quotation mark
+            '\u201E'  // low double left quote used in german
+        };
+
+        /// <summary>
+        /// Returns completion results for PropertyType parameter.
+        /// </summary>
+        /// <param name="commandName">The command name.</param>
+        /// <param name="parameterName">The parameter name.</param>
+        /// <param name="wordToComplete">The word to complete.</param>
+        /// <param name="commandAst">The command AST.</param>
+        /// <param name="fakeBoundParameters">The fake bound parameters.</param>
+        /// <returns>List of Completion Results.</returns>
+        public IEnumerable<CompletionResult> CompleteArgument(
+            string commandName,
+            string parameterName,
+            string wordToComplete,
+            CommandAst commandAst,
+            IDictionary fakeBoundParameters)
+        {
+            if (!IsRegistryProvider(fakeBoundParameters))
+            {
+                yield break;
+            }
+
+            var propertyTypePattern = WildcardPattern.Get(wordToComplete.Trim(QuoteChars) + "*", WildcardOptions.IgnoreCase);
```

Comment:
There is an internal method in CompletionCompleters called `HandleDoubleAndSingleQuote` that you can use instead of manually handling it here. It also outputs the quote type that the user entered, allowing you to add it back to the completion result if you want (I don't do it in my module because I prefer to only have quotes if they are needed).

Reply:
Thanks @MartinGC94. I didn't realise this method existed. I've added it here: 2af476ec5. Let me know if you want anything else done to it, I add the quotes back into the completion text if they existed in the first place.